version: '3.8'
services:
  discovery-service:
    container_name: discovery
    build:
      context: .
      dockerfile: ./service-discovery/Dockerfile
    ports:
      - "8761:8761"
    healthcheck:
      test: "curl --fail --silent localhost:8761/actuator/health | grep UP || exit 1"
      interval: 20s
      timeout: 5s
      retries: 5
      start_period: 40s

  api-gateway:
    container_name: api-gateway
    build:
      context: .
      dockerfile: ./api-gateway/Dockerfile
    depends_on:
      discovery-service:
        condition: service_healthy
    ports:
      - "80:80"
  users-db:
    container_name: users-db
    image: postgres
    environment:
      POSTGRES_DB: users
      POSTGRES_USER: users_admin
      POSTGRES_PASSWORD: users_pass
      PGDATA: /data/users
    volumes:
      - postgres:/data/users
    ports:
      - "5432:5432"
    networks:
      - postgres
    restart: unless-stopped

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL:-pgadmin4@pgadmin.org}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD:-admin}
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
      - pgadmin:/var/lib/pgadmin
    ports:
      - "5050:80"
    networks:
      - postgres
    restart: unless-stopped

networks:
  postgres:
    driver: bridge
volumes:
  postgres:
  pgadmin: